src/lib/web-component/log-view/log-view-element.js

```diff
@@ -51,45 +103,102 @@ export class LogView extends HTMLElement {
     this.tableBody.appendChild(rowsFragment);
   }
 
+  handleEvent(event) {
+    if (event.type === 'click') {
+      const logDetails = event.target.closest('tr')?._log;
+
+      if (logDetails) {
+        this.openDetailSidebar(logDetails);
+        return;
+      }
+
+      switch (event.target) {
+        case this.closeBtn:
+          this.closeDetailSidebar();
+          break;
+        case this.clearTSFilterBtn:
+          this.hideTSFilter();
+          break;
+        default:
+          throw new Error(`wrong event target found: ${event.target.tagName}`);
+      }
+    } else if (event.type === 'contextmenu') {
+      browser.menus.overrideContext({});
+      this.createContextMenu();
+      browser.menus.refresh();
+    }
+  }
+
   openDetailSidebar(logDetails) {
     const logString = JSON.stringify(logDetails);
     this.logDetails.textContent = logString;
     this.logTableWrapper.classList.add('width-60');
+    this.timeStampFilterContainer.classList.add('width-60');
     this.logDetailWrapper.hidden = false;
   }
 
   closeDetailSidebar() {
     this.logDetailWrapper.hidden = true;
     this.logTableWrapper.classList.remove('width-60');
+    this.timeStampFilterContainer.classList.remove('width-60');
   }
 
-  handleEvent(event) {
-    if (event.type === 'click') {
-      const logDetails = event.target.closest('tr')?._log;
+  hideTSFilter() {
+    this.timeStamp.startTime = null;
+    this.timeStamp.stopTime = null;
+    this.disptachTSFilterEvent();
 
-      if (logDetails) {
-        this.openDetailSidebar(logDetails);
-        return;
-      }
+    this.startTimeLabel.textContent = 'From Beginning';
+    this.stopTimeLabel.textContent = 'Up to End';
+    this.timeStampFilterContainer.hidden = true;
+  }
 
-      if (event.target === this.closeBtn) {
-        this.closeDetailSidebar();
-      }
-    }
+  createContextMenu() {
+    browser.menus.create({
+      id: 'startTime',
+      title: 'Start from this timestamp',
+      contexts: ['page'],
+      viewTypes: ['tab'],
+      documentUrlPatterns: [
+        `moz-extension://${location.host}/activitylog/activitylog.html`,
```

Comment:
Use `location.href` instead. Otherwise the menu won't show up if the page has a query string.

Reply:
done.